### PR TITLE
move `parse_cfgspecs` to `rustc_interface`

### PR DESCRIPTION
**Pull request by Centril**
Part of https://github.com/rust-lang/rust/pull/65324.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/librustc_interface/tests.rs [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-    build_session_options,
-    to_crate_config,
-    parse_cfgspecs,
-};
-use crate::session::config::{LtoCli, LinkerPluginLto, SwitchWithOptPath, ExternEntry};
-use crate::session::build_session;
-use crate::session::search_paths::SearchPath;
+extern crate getopts;
+
+use crate::interface::parse_cfgspecs;
```
Comment (Centril) [inline src/librustc_interface/tests.rs]: (This file has to be moved because of this dependency.)

**Comment by Centril** [comment]
cc @petrochenkov

**Comment by Mark-Simulacrum** [comment]
r=me if this strictly moves the code (no changes) -- it's practically impossible to tell if this is the case, though it seems to be.

**Comment by Mark-Simulacrum** [comment]
For the future the best way I know of to make this sort of change reviewable is to have one commit which prepares and is strictly a move and then a second commit which fixes up the previous one -- this way the diff can be reviewed as strictly the fixup changes.

**Pull request merged**